code/datums/ammo/rocket.dm

```diff
@@ -62,62 +62,65 @@
-		M.ex_act(300, P.dir, P.weapon_cause_data, 100)
-	cell_explosion(T, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, P.weapon_cause_data)
-	smoke.set_up(1, T)
+
+/datum/ammo/rocket/ap/on_hit_mob(mob/mob, obj/projectile/projectile)
+	var/turf/turf = get_turf(mob)
+	mob.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
+	mob.apply_effect(3, WEAKEN)
+	mob.apply_effect(3, PARALYZE)
+	if(ishuman_strict(mob)) // No yautya or synths. Makes humans gib on direct hit.
+		mob.ex_act(300, projectile.dir, projectile.weapon_cause_data, 100)
+	cell_explosion(turf, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	smoke.set_up(1, turf)
 	smoke.start()
 
-/datum/ammo/rocket/ap/on_hit_obj(obj/O, obj/projectile/P)
-	var/turf/T = get_turf(O)
-	O.ex_act(150, P.dir, P.weapon_cause_data, 100)
-	cell_explosion(T, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, P.weapon_cause_data)
-	smoke.set_up(1, T)
+/datum/ammo/rocket/ap/on_hit_obj(obj/object, obj/projectile/projectile)
+	var/turf/turf = get_turf(object)
+	object.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
+	cell_explosion(turf, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	smoke.set_up(1, turf)
 	smoke.start()
 
-/datum/ammo/rocket/ap/on_hit_turf(turf/T, obj/projectile/P)
+/datum/ammo/rocket/ap/on_hit_turf(turf/turf, obj/projectile/projectile)
 	var/hit_something = 0
-	for(var/mob/M in T)
-		M.ex_act(150, P.dir, P.weapon_cause_data, 100)
-		M.apply_effect(4, PARALYZE)
+	for(var/mob/mob in turf)
+		mob.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
+		mob.apply_effect(3, WEAKEN)
+		mob.apply_effect(3, PARALYZE)
 		hit_something = 1
 		continue
 	if(!hit_something)
-		for(var/obj/O in T)
-			if(O.density)
-				O.ex_act(150, P.dir, P.weapon_cause_data, 100)
+		for(var/obj/object in turf)
+			if(object.density)
+				object.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
 				hit_something = 1
 				continue
 	if(!hit_something)
-		T.ex_act(150, P.dir, P.weapon_cause_data, 200)
+		turf.ex_act(150, projectile.dir, projectile.weapon_cause_data, 200)
 
-	cell_explosion(T, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, P.weapon_cause_data)
-	smoke.set_up(1, T)
+	cell_explosion(turf, 100, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	smoke.set_up(1, turf)
 	smoke.start()
 
-/datum/ammo/rocket/ap/do_at_max_range(obj/projectile/P)
-	var/turf/T = get_turf(P)
+/datum/ammo/rocket/ap/do_at_max_range(obj/projectile/projectile)
+	var/turf/turf = get_turf(projectile)
 	var/hit_something = 0
-	for(var/mob/M in T)
-		M.ex_act(250, P.dir, P.weapon_cause_data, 100)
-		M.apply_effect(2, WEAKEN)
-		M.apply_effect(2, PARALYZE)
+	for(var/mob/mob in turf)
+		mob.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
+		mob.apply_effect(3, WEAKEN)
+		mob.apply_effect(3, PARALYZE)
 		hit_something = 1
 		continue
```

Comment:
this should be a break not a continue otherwise it'll explode every mob in the contents, same below with objects, that's not consistent behavior with the mutually exclusive hit checks and in on_hit_turf etc etc